### PR TITLE
Add simple log processor implementation and log exporter interface

### PR DESCRIPTION
**Pull request by kxyr**
This is part of the Logging SDK for # 337, following the Logging SDK Initial Commit in # 386.

This PR adds:

* A simple processor implementation and unit tests
* A log exporter interface, required for the processor unit tests 

The simple log processor sends Log Records from the SDK directly to an exporter in a batch of one. Its design is similar to the simple span processor for traces.

The design for these components will be covered in a future PR with the combined Logging SDK design document. 

cc @MarkSeufert  @alolita

### CODE REVIEW

File: sdk/include/opentelemetry/sdk/logs/exporter.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+{
+public:
+  virtual ~LogExporter() = default;
+
+  /**
+   * Format each LogRecord as a (possibly dimensionally altered) LogRecord, then
+   * send the LogRecord to the exporter destination.
+   *
+   * The default timeout is set to 0 if unspecified.
+   * @param records: a vector of shared pointers to log records
```
Comment (ranton256) [inline sdk/include/opentelemetry/sdk/logs/exporter.h]: comment says shared_ptr but param is unique_ptr

File: sdk/include/opentelemetry/sdk/logs/simple_log_processor.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ */
+class SimpleLogProcessor : public LogProcessor
+{
+
+public:
+  /**
+   * Initialize a simple log processor.
+   * @param exporter the exporter used by the log processor
+   */
+  //  // constructor and destructor
```
Comment (ranton256) [inline sdk/include/opentelemetry/sdk/logs/simple_log_processor.h]: nit: extra //

File: sdk/src/logs/simple_log_processor.cc [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  std::vector<std::unique_ptr<opentelemetry::logs::LogRecord>> batch;
+  batch.emplace_back(std::move(record));
+
+  // get lock to ensure while Export() is never called concurrently
+  std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock_);
+
+  // check the ExportCode returned from the export
+  if (exporter_->Export(batch) == ExportResult::kFailure)
+  {
+    /* TODO: throw exception? */
```
Comment (ranton256) [inline sdk/src/logs/simple_log_processor.cc]: signature is noexcept though, so probably need to do something else, void return and noexcept and logging already failed so unless the API changes, that basically leaves doing nothring or printing to cerr/stderr as only options.

Comment (ranton256) [reply sdk/src/logs/simple_log_processor.cc]: Do exporters have any means for saving state or metrics on failures? That might be something to discuss with maintainers, what is strategy when log exporter fails.

**Review by alolita — APPROVED** [review]
Ask Reiley for clear guidance on exception handling for logging - should we just raise an exception, dump to stderr or to a log? What does the spec say?

**Pull request closed without merging**